enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,498 @@
+service on the host to apply the provided NMState files. This is the approach
+the [prototype implementation](https://docs.google.com/document/d/1Zp1J2HbVvu-v4mHc9M594JLqt6UwdKoix8xkeSzVL98/edit#heading=h.fba4j9nvp0nl)
+used since per-host configuration was not available any other way. However,
+this feels messy as it requires every node to have the configuration for every
+other node in the cluster and select the correct one for itself. It does have
+the benefit of requiring significantly less work to implement. If users are
+writing their NMState configs in machine-configs instead of via a top-level
+OpenShift API perhaps this also side-steps the concerns about API style
+compatibility between OpenShift and NMState?
+
+### API Extensions
+
+This will not directly modify the API of OpenShift. The API for this feature is
+implemented in Kubernetes-NMState, and this does not require any specific
+modifications in that project at this time.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+Because this gives the deployer a great deal of control over the bridge
+configuration for OVNK, they will have the ability to configure it incorrectly,
+which may cause problems either on initial deployment or at any point after.
+We will want to ensure that must-gather is collecting enough information about
+the bridge configuration for us to determine whether it is compliant with our
+guidelines, and we must have clear instructions on what is required for br-ex.
+
+### Drawbacks
+
+OpenShift has historically avoided per-host configuration for cluster nodes.
+This is a large departure from that philosophy, but that is unavoidable.
+Certain aspects of networking (notably static IPs) are by nature host-specific
+and cannot be handled any other way.
+
+
+## Design Details
+
+### Open Questions [optional]
+
+* There has been resistance to using the NMState API in OpenShift in the past
+  because it is not compliant with the OpenShift API Guidelines. There is work
+  underway in NMState to address this, but is that mandatory for this to be
+  implemented?
+
+* Which operator is going to be responsible for deploying the NMState configuration?
+  MCO will need to be involved since it needs to be included in Ignition, but
+  do we want to use machine-configs? They can't be modified on day 2 without
+  the machine config pool being degraded, but since day 2 modifications would
+  be done via Kubernetes-NMState maybe that's okay? We'll apply them once and
+  never touch them again.
+
+* If changes are made to the bridge configuration on day 2, will OVNKubernetes
+  handle those correctly without a reboot? If not, how do we orchestrate a
+  reboot?
```

Comment:
OVN-Kubernetes will not handle the replacement of the gateway interface (`enp2s0` in the example above). This is an existing limitation where OVN-Kubernetes installs an OpenFlow rule in br-ex that states the gateway interface is the OVS port ID of `enp2s0` (port 1) and where attaching a new interface, say `enp3s0`, its OVS port ID will be different hence no egress traffic forwarding.

Monitoring br-ex port attachments and recalculating the OpenFlow rule may help. This is outside the scope of this enhancement, though.

Reply:
Ack, thanks. I'm just going to include your comment in the enhancement.

Reply:
I'm wondering if that's because configure-ovs uses different profile names from NMState. It's possible we ended up with multiple profiles doing the same thing.

Maybe I just need to test this. I can deploy my prototype and change the MTU or something like that and see what happens.

Reply:
At the very least I recall nmstate changing the names of things, like ovs ports and such. Yeah, I am curious about testing this again.